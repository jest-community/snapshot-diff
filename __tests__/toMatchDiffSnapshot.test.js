// @flow

const snapshotDiff = require('../src/index');

const a = `
    some
    some
    some
    some
    some
    some
    some
    not
    very
    long
    script
`;
const b = `
    some
    some
    some
    some
    some
    some
    some
    not
    so
    very
    long
    script
`;

beforeAll(() => {
  expect.extend({ toMatchDiffSnapshot: snapshotDiff.toMatchDiffSnapshot });
});

test('works with default options', () => {
  // $FlowFixMe
  expect(a).toMatchDiffSnapshot(b);
});

[
  { expand: true },
  { colors: true },
  { contextLines: 0 },
].forEach((options: any) => {
  test(`proxies "${Object.keys(options).join(', ')}" option(s)`, () => {
    // $FlowFixMe
    expect(a).toMatchDiffSnapshot(b, options);
  });
});
