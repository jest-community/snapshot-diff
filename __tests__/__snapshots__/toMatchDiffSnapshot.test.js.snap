// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proxies "colors" option(s) 1`] = `
"Snapshot Diff:
[32m- First value[39m
[31m+ Second value[39m

[33m@@ -5,8 +5,9 @@[39m
[33m[39m[2m     some[22m
[2m     some[22m
[2m     some[22m
[2m     some[22m
[2m     not[22m
[31m+    so[39m
[2m     very[22m
[2m     long[22m
[2m     script[22m"
`;

exports[`proxies "contextLines" option(s) 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -10,0 +10,1 @@
+    so"
`;

exports[`proxies "expand" option(s) 1`] = `
"Snapshot Diff:
- First value
+ Second value

  
      some
      some
      some
      some
      some
      some
      some
      not
+     so
      very
      long
      script"
`;

exports[`works with default options 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -5,8 +5,9 @@
     some
     some
     some
     some
     not
+    so
     very
     long
     script"
`;
